### PR TITLE
MIOpen bf16 batchnorm enablement

### PR DESCRIPTION
**Pull request by jithunnair-amd**

### CODE REVIEW

**Comment by rocm-mici** [comment]
Jenkins build for [77d8e922332f6ae3ab04a1c4dfc88196303949fb](https://github.com/ROCm/pytorch/commit/77d8e922332f6ae3ab04a1c4dfc88196303949fb) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/1//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/1/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [e8483eacd72b5ff5c227254f6acda6422c5590b3](https://github.com/ROCm/pytorch/commit/e8483eacd72b5ff5c227254f6acda6422c5590b3) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/2//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/2/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [e8483eacd72b5ff5c227254f6acda6422c5590b3](https://github.com/ROCm/pytorch/commit/e8483eacd72b5ff5c227254f6acda6422c5590b3) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/3//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/3/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/4//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/4/artifact)

**Comment by rocm-mici** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/5//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/5/artifact)

Detected error during base docker image building: 
```
#29 14.18 The following packages have unmet dependencies:
#29 14.25  rocm-dev : Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 14.25             Depends: rocm-device-libs (= 1.0.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 14.25  rocm-utils : Depends: rocminfo (= 1.0.0.60203-124~22.04)
#29 14.25               Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 14.25 E: Unable to correct problems, you have held broken packages.
#29 ERROR: process "/bin/sh -c bash ./install_rocm.sh" did not complete successfully: exit code: 100
------
 > [24/49] RUN bash ./install_rocm.sh:
14.18 distribution that some required packages have not yet been created
14.18 or been moved out of Incoming.
```

**Comment by rocm-mici** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/6//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/6/artifact)

Detected error during base docker image building: 
```
#29 16.15 The following packages have unmet dependencies:
#29 16.23  rocm-dev : Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 16.23             Depends: rocm-device-libs (= 1.0.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 16.23  rocm-utils : Depends: rocminfo (= 1.0.0.60203-124~22.04)
#29 16.23               Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 16.24 E: Unable to correct problems, you have held broken packages.
#29 ERROR: process "/bin/sh -c bash ./install_rocm.sh" did not complete successfully: exit code: 100
------
 > [24/49] RUN bash ./install_rocm.sh:
16.15 distribution that some required packages have not yet been created
16.15 or been moved out of Incoming.
```

**Comment by rocm-mici** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/8//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/8/artifact)

Detected error during base docker image building: 
```
#29 12.21 The following packages have unmet dependencies:
#29 12.27  rocm-dev : Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 12.27             Depends: rocm-device-libs (= 1.0.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 12.27  rocm-utils : Depends: rocminfo (= 1.0.0.60203-124~22.04)
#29 12.27               Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 12.28 E: Unable to correct problems, you have held broken packages.
#29 ERROR: process "/bin/sh -c bash ./install_rocm.sh" did not complete successfully: exit code: 100
------
 > [24/49] RUN bash ./install_rocm.sh:
12.21 distribution that some required packages have not yet been created
12.21 or been moved out of Incoming.
```

**Comment by okakarpa** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/9//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/9/artifact)

Detected error during base docker image building: 
```
#29 11.30 The following packages have unmet dependencies:
#29 11.36  rocm-dev : Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.36             Depends: rocm-device-libs (= 1.0.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.36  rocm-utils : Depends: rocminfo (= 1.0.0.60203-124~22.04)
#29 11.36               Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.36 E: Unable to correct problems, you have held broken packages.
#29 ERROR: process "/bin/sh -c bash ./install_rocm.sh" did not complete successfully: exit code: 100
------
 > [24/49] RUN bash ./install_rocm.sh:
11.30 distribution that some required packages have not yet been created
11.30 or been moved out of Incoming.
```

**Comment by okakarpa** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/10//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/10/artifact)

Detected error during base docker image building: 
```
#29 11.92 The following packages have unmet dependencies:
#29 11.98  rocm-dev : Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.98             Depends: rocm-device-libs (= 1.0.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.98  rocm-utils : Depends: rocminfo (= 1.0.0.60203-124~22.04)
#29 11.98               Depends: rocm-cmake (= 0.13.0.60203-124~22.04) but 5.0.0-1 is to be installed
#29 11.99 E: Unable to correct problems, you have held broken packages.
#29 ERROR: process "/bin/sh -c bash ./install_rocm.sh" did not complete successfully: exit code: 100
------
 > [24/49] RUN bash ./install_rocm.sh:
11.92 distribution that some required packages have not yet been created
11.92 or been moved out of Incoming.
```

**Comment by rocm-mici** [comment]
Jenkins build for [f0effc8b8d2f4e71f58dbe3a12880d68021ea995](https://github.com/ROCm/pytorch/commit/f0effc8b8d2f4e71f58dbe3a12880d68021ea995) commit finished as FAILURE
Links: [Blue Ocean view](http://ml-ci-internal.amd.com:8080/blue/organizations/jenkins/pytorch%2Fpytorch-ci-pipeline/detail/PR-1665/11//pipeline) / [Build artifacts](http://ml-ci-internal.amd.com:8080/job/pytorch/job/pytorch-ci-pipeline/job/PR-1665/11/artifact)
